### DISCUSSION TITLE
EntityDTO maps one-to-many relationships to arrays of strings

### DISCUSSION THREAD

**Discussion by akapps** [mikro-orm/mikro-orm#5837] [General]
### Describe the bug

Migrating to MikroORM v.6.3.0 fails with TypeScript compilation errors.
Our project is largely relying on `EntityDTO` types, which now appear to mistakenly transform all `Collection<Entity>` types into `string[]`.

This is new to MikroORM 6.3.0 - and I couldn't find any mention to such a change in the changelog.
Also, objects at runtime are indeed the ones I expect, not strings.

It pretty much looks like a bug with the computed type to me - or am I missing something obvious ?

### Reproduction

Define a very simple one-to-many relationship:

```ts
@Entity()
export class Book {
  @PrimaryKey()
  id: string;

  @Property()
  name: string;

  @ManyToOne(() => Author)
  author: Author;
}

@Entity()
export class Author {
  @OneToMany(() => Book, (book) => book.author)
  books = new Collection<Book>(this);
}
```

Accessing `author.books` on an `EntityDTO` fails TS compilation:

```ts
function doSomething(author: EntityDTO<Author>) {
  if (author.books.length) {
    const [firstBook] = author.books;
    return firstBook.name;
  }
}
```

from TS compiler:

```
src/author.entity.ts:41:22 - error TS2339: Property 'name' does not exist on type 'string'.

41     return firstBook.name;

```

_**Note:** I noted that if I remove the `Book.id` property (or rename it), computed `EntityDTO` type is back to normal._

**Comment by B4nan** [comment]
You need to have correct populate hint on type level. `EntityDTO<Author>` means no relations are populated, you need `EntityDTO<Loaded<Author, 'books'>>`.

**Reply by akapps** [reply]
Got it. We'll have to redefine a few of our types, but it's actually stricter - and safer.

Thank you for your quick answer.
